### PR TITLE
[Feature] add model script of cait

### PR DESCRIPTION
**Pull request by wcrzlh**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

The model script of cait is added.

## Test Plan

Please use 'create_model' part for testing models in cait.

## Related Issues and PRs

Related issue: https://github.com/mindspore-lab/mindcv/issues/424

### CODE REVIEW

File: mindcv/models/cait.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        return x
+
+    def construct(self, x: Tensor) -> Tensor:
+        x = self.forward_features(x)
+        x = self.forward_head(x)
+        return x
+
+
+@register_model
+def cait_XXS24_224(pretrained: bool = False, num_classes: int = 1000, in_channels=3, **kwargs) -> CaiT:
```
Comment (geniuspatrick) [inline mindcv/models/cait.py:346]: 这个函数名也是

**Review by geniuspatrick — CHANGES_REQUESTED** [review]

File: mindcv/models/cait.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+
+
+def _cfg(url='', **kwargs):
+    return {
+        'url': url,
+        'num_classes': 1000,
+        'first_conv': '', 'classifier': '',
+        **kwargs
+    }
+
```
[the remark this answers is not in the document]
Comment (wcrzlh) [reply mindcv/models/cait.py:40]: added

File: mindcv/models/cait.py [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+
+class Class_Attention(nn.Cell):
```
[the remark this answers is not in the document]
Comment (wcrzlh) [reply mindcv/models/cait.py:53]: 是的

File: mindcv/models/cait.py [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+    def construct(self, x) -> Tensor:
+        B, N, C = x.shape
+        qkv = ops.reshape(self.qkv(x), (B, N, 3, self.num_heads, C // self.num_heads))
+        qkv = ops.transpose(qkv, (2, 0, 3, 1, 4))
+        q, k, v = ops.unstack(qkv, axis=0)
+        q = ops.mul(q, self.scale)
+
+        attn = self.q_matmul_k(q, k)
+
+        "talking head trick"
```
[the remark this answers is not in the document]
Comment (wcrzlh) [reply mindcv/models/cait.py]: It is a trick used in cait. I have deleted this comment line.

File: mindcv/models/cait.py [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+        self.mlp = Mlp(in_features=dim,
+                       hidden_features=int(dim * mlp_ratio),
+                       drop=drop_rate,
+                       act_layer=act_layer)
+
+        self.gamma_1 = Parameter(init_values * ops.ones((dim), ms.float32), requires_grad=True)
+        self.gamma_2 = Parameter(init_values * ops.ones((dim), ms.float32), requires_grad=True)
+
+    def construct(self, x: Tensor) -> Tensor:
+
```
[the remark this answers is not in the document]
Comment (wcrzlh) [reply mindcv/models/cait.py]: deleted

File: mindcv/models/cait.py [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+        self.norm = norm_layer((embed_dim,))
+
+        self.head = nn.Dense(embed_dim, num_classes) if num_classes > 0 else nn.Identity()
+
+        self.pos_embed = init.initializer(TruncatedNormal(sigma=0.02), self.pos_embed.shape, ms.float32)
+        self.cls_token = init.initializer(TruncatedNormal(sigma=0.02), self.cls_token.shape, ms.float32)
+
+        self._init_weights()
+
+    def _init_weights(self) -> None:
```
[the remark this answers is not in the document]
Comment (wcrzlh) [reply mindcv/models/cait.py]: fixed

File: mindcv/models/cait.py (hunk 1 above)
Comment (wcrzlh) [reply mindcv/models/cait.py:355]: 原作者实现中写的就是这样的函数名

**Review by geniuspatrick — APPROVED** [review]

**Pull request merged**